CONTRIBUTING.md

```diff
@@ -52,6 +52,20 @@ Check out the [security policy](https://github.com/NixOS/nix/security/policy).
 
      Link related issues to inform interested parties and future contributors about your change.
      If your pull request closes one or multiple issues, mention that in the description using `Closes: #<number>`, as it will then happen automatically when your change is merged.
+   * Credit original authors when you're reusing or building on their work.
+   * Link to relevant changes in other projects, so that others can understand the full context of the change in the future when you or someone else will change or troubleshoot the code.
+     This is especially important when your change is based on work done in other repositories.
+
+     Example:
+     ```
+     This is based on the work of @user in <url>.
+     This solution took inspiration from <url>.
+
+     Co-authored-by: User Name <user@example.com>
+     ```
+
+     Use the `git cherry-pick -x` flag, and amend the commits to link to forks when applicable.
```

Comment:
So far I was using the git commit url since the commit reported by cherry-pick, would not be valid in nix at any point or am I missing something?

Reply:
Would this be an improvement?

```suggestion
     When cherry-picking from a different repository, use the `-x` flag, and then amend the commits to turn the hashes into URLs.
```

Reply:
Yes. This is better. Now I understand.